subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -1200,9 +1200,10 @@ static bool dhcpv4_parse_options(struct net_pkt *pkt,
 			log_server.sin_family = AF_INET;
 			log_backend_net_set_ip((struct sockaddr *)&log_server);
 
-#ifdef CONFIG_LOG_BACKEND_NET_AUTOSTART
-			log_backend_net_start();
-#endif
+			if (IS_ENABLED(CONFIG_LOG_BACKEND_NET_AUTOSTART) &&
+			    !IS_ENABLED(CONFIG_NET_CONFIG_SETTINGS)) {
+				log_backend_net_start();
+			}
```

Comment:
Not sure how this is related to "avoiding automatic enabling"
Is there now two separate changes in this commit, one for automatic enabling and one for ACD?

Reply:
split the commit in two.